Packs/Kafka/Integrations/KafkaV3/KafkaV3.py

```diff
@@ -0,0 +1,729 @@
+            if polled_msg:
+                demisto.debug("Received a message from Kafka.")
+                incidents.append(create_incident(message=polled_msg, topic=topic))
+                last_fetched_offsets[f'{polled_msg.partition()}'] = polled_msg.offset()
+
+    finally:
+        if kafka_consumer:
+            kafka_consumer.close()
+
+    last_run = {'last_fetched_offsets': last_fetched_offsets, 'last_topic': topic}
+    demisto.debug(f"Fetching finished, setting last run to {last_run}")
+    demisto.setLastRun(last_run)
+
+    demisto.incidents(incidents)
+
+
+''' COMMANDS MANAGER / SWITCH PANEL '''
+
+
+def main():  # pragma: no cover
+    demisto_command = demisto.command()
+    demisto_params = demisto.params()
+    demisto_args = demisto.args()
+    demisto.debug(f'Command being called is {demisto_command}')
+    brokers = demisto_params.get('brokers')
+    offset = handle_empty(demisto_params.get('offset', 'earliest'), 'earliest')
+    trust_any_cert = demisto_params.get('insecure', False)
+
+    # Should we use SSL
+    use_ssl = demisto_params.get('use_ssl', False)
+
+    if use_ssl:
+        # Add Certificates
+        ca_cert = demisto_params.get('ca_cert', None)
+        client_cert = demisto_params.get('client_cert', None)
+        client_cert_key = demisto_params.get('client_cert_key', None)
+        ssl_password = demisto_params.get('additional_password', None)
+        kafka_kwargs = {'brokers': brokers, 'ca_cert': ca_cert, 'client_cert': client_cert,
+                        'client_cert_key': client_cert_key, 'ssl_password': ssl_password, 'offset': offset,
+                        'trust_any_cert': trust_any_cert}
+    else:
+        kafka_kwargs = {'brokers': brokers, 'offset': offset, 'trust_any_cert': trust_any_cert}
+
+    kafka = KafkaCommunicator(**kafka_kwargs)
+
+    try:
+        if demisto_command == 'test-module':
+            return_results(command_test_module(kafka, demisto_params))
+        elif demisto_command == 'kafka-print-topics':
+            return_results(print_topics(kafka, demisto_args))
+        elif demisto_command == 'kafka-publish-msg':
+            produce_message(kafka, demisto_args)
+        elif demisto_command == 'kafka-consume-msg':
+            return_results(consume_message(kafka, demisto_args))
+        elif demisto_command == 'kafka-fetch-partitions':
+            return_results(fetch_partitions(kafka, demisto_args))
+        elif demisto_command == 'fetch-incidents':
+            fetch_incidents(kafka, demisto_params)
+        else:
+            raise NotImplementedError(f'Command {demisto_command} not found in command list')
+
+    except Exception as e:
+        debug_log = ''
+        stacktrace = traceback.format_exc()
+        if stacktrace:
+            debug_log += f'Debug logs:\nFull stacktrace:\n\n{stacktrace}'
+        return_error(f'{str(e)}\n\n{debug_log}')
+
+    finally:
+        if os.path.isfile('ca.cert'):
```

Comment:
wouldnt it be better to refer to the paths stored in the `KafkaCommunicator` instance ?